elide-core/src/test/java/com/yahoo/elide/jsonapi/JsonApiTest.java

```diff
@@ -326,5 +339,21 @@ public void readListWithMeta() throws IOException {
         assertEquals(attributes.get("firstName"), "bob");
         assertEquals(data.getRelationships().get("children").getData().getSingleValue().getId(), "2");
         assertNull(included);
+        checkEquality(jsonApiDocument);
+    }
+
+    private void checkEquality(JsonApiDocument doc1) {
+        JsonApiDocument doc2;
+        try {
+            String json = mapper.writeJsonApiDocument(doc1);
+            doc2 = mapper.readJsonApiDocument(json);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        if (doc2.getData() == null) {
```

Comment:
Why do we do this?

Reply:
Good question.  Seems the null is not handled well.

I found the problem. Somehow serializing changes the value of doc.data!

Reply:
I removed this hack and added a verification that data is unchanged after serialization, which fails :(

Reply:
It seems "Empty" object and Null object are written the same, but read differently.

Reply:
expected: 
<JsonApiDocument(data=Data(values=[], relationshipType=MANY_TO_ONE), meta=null, links=null, includedRecs=[], included=null)> 
but was: 
<JsonApiDocument(data=null, meta=null, links=null, includedRecs=[], included=null)>

Reply:
Will now treat "Empty" object and Null data object as equal.